### PR TITLE
feat(core): add new address for customer

### PR DESCRIPTION
**Pull request by bc-alexsaiannyi**
## What/Why?
This PR brings functionality for adding new address for customer.

## Testing
locally

Screenshot 2024-05-21 at 12 06 11

Screenshot 2024-05-21 at 12 06 40

Screenshot 2024-05-21 at 12 07 00

### CODE REVIEW

File: core/app/[locale]/(default)/account/[tab]/_components/add-address/handlers.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+    provinceSetter(states ?? []);
+  };
+
+export {
+  createTextInputValidationHandler,
+  createPicklistOrTextValidationHandler,
+  createPasswordValidationHandler,
+  createCountryChangeHandler,
+};
```
Comment (bc-alexsaiannyi) [inline core/app/[locale]/(default)/account/[tab]/_components/add-address/handlers.ts:81]: NOTE: I've put handlers separately to reuse them in upcoming `<updateAddressForm />`. Also after covering functionality for different account-related  forms we can re-use them in other forms as well.

File: core/app/[locale]/(default)/account/[tab]/_components/addresses-content/add-address.tsx [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+    submit: string;
+    submitting: string;
+  };
+}
+
+const SubmitButton = ({ messages }: SumbitMessages) => {
+  const { pending } = useFormStatus();
+
+  return (
+    <Button
```
Comment (jorgemoya) [inline core/app/[locale]/(default)/account/[tab]/_components/addresses-content/add-address.tsx:55]: We have updated the Button component to have a `loading` state. Should use that instead.

File: core/app/[locale]/(default)/account/[tab]/_components/addresses-content/add-address.tsx [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+              )}
+            </Field>
+          )}
+        </div>
+
+        <div className="mt-8 flex flex-col justify-stretch gap-2 md:flex-row md:justify-start md:gap-6">
+          <FormSubmit asChild>
+            <SubmitButton messages={{ submit: t('submit'), submitting: t('submitting') }} />
+          </FormSubmit>
+          <Button className="items-center px-8 md:w-fit" variant="secondary">
```
Comment (jorgemoya) [inline core/app/[locale]/(default)/account/[tab]/_components/addresses-content/add-address.tsx:244]: Should this Slot in the Link?

File: core/app/[locale]/(default)/account/[tab]/_components/addresses-content/address-book.tsx [hunk 4]
```diff
@@ -127,7 +126,9 @@ export const AddressesList = ({ customerAddressBook, addressesCount }: Props) =>
         )}
         <li className="flex w-full border-collapse flex-col justify-start gap-2 border-t border-gray-200 pt-8">
           <Button aria-label={t('addNewAddress')} asChild className="w-fit hover:text-white">
```
Comment (jorgemoya) [inline core/app/[locale]/(default)/account/[tab]/_components/addresses-content/address-book.tsx:128]: Same here.

File: core/app/[locale]/(default)/login/register-customer/_components/register-customer-form/index.tsx [hunk 5]
```diff
@@ -49,52 +54,13 @@ interface RegisterCustomerProps {
...
   };
 }
 
-const createFieldName = (fieldType: 'customer' | 'address', fieldId: number) => {
-  const secondFieldType = fieldType === 'customer' ? 'address' : 'customer';
-
-  return `${fieldType}-${BOTH_CUSTOMER_ADDRESS_FIELDS.includes(fieldId) ? `${secondFieldType}-` : ''}${FieldNameToFieldId[fieldId] || fieldId}`;
-};
-
 const SubmitButton = ({ messages }: SumbitMessages) => {
```
Comment (jorgemoya) [inline core/app/[locale]/(default)/login/register-customer/_components/register-customer-form/index.tsx:64]: Can you change this button too to use new `loading` state?

File: core/app/[locale]/(default)/account/[tab]/_components/addresses-content/add-address.tsx (hunk 3 above)
Comment (bc-alexsaiannyi) [reply core/app/[locale]/(default)/account/[tab]/_components/addresses-content/add-address.tsx]: Good catch, I've missed it.

**Review by yurytut1993 — APPROVED** [review]
👌

**Review by BC-krasnoshapka — APPROVED** [review]

**Pull request merged**